pkgs/dart_mcp/lib/src/api/completions.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'api.dart';
+
+/// A request from the client to the server, to ask for completion options.
+extension type CompleteRequest.fromMap(Map<String, Object?> _value)
+    implements Request {
+  static const methodName = 'completion/complete';
+
+  factory CompleteRequest({
+    required Reference ref,
+    required CompletionArgument argument,
+    MetaWithProgressToken? meta,
+  }) => CompleteRequest.fromMap({
+    'ref': ref,
+    'argument': argument,
+    if (meta != null) '_meta': meta,
+  });
+
+  /// A reference to the thing to complete.
```

Comment:
Can you expand this?

Reply:
Added a link to the specific reference types, not sure what else to put. This is undocumented in the schema.